wallet/bench/Client/Pos/Wallet/Web/Api.hs

```diff
@@ -0,0 +1,52 @@
+{-# LANGUAGE TypeOperators #-}
+
+-- | Client functions for Wallet Web API. To be more precise,
+-- only for endpoints we have to benchmark.
+
+module Client.Pos.Wallet.Web.Api
+    ( getHistory
+    , getWallets
+    , newPayment
+    ) where
+
+import           Universum
+
+import           Servant.API                ((:<|>) (..), (:>))
+import           Servant.Client             (ClientM, client)
+
+import           Pos.Client.Txp.Util        (InputSelectionPolicy)
+import           Pos.Core.Types             (Coin)
+import           Pos.Wallet.Web.Api         (ApiPrefix, GetHistory, GetWallets,
+                                             NewPayment)
+import           Pos.Wallet.Web.ClientTypes (Addr, CAccountId (..),
+                                             CId (..), CPassPhrase, CTx, CWallet,
+                                             ScrollLimit, ScrollOffset, Wal)
+
+-- | "Benchmarking API" which includes
+-- endpoints we need for benchmarking.
+type WalletBenchApi = ApiPrefix :> (
+     GetHistory
+    :<|>
+     GetWallets
+    :<|>
+     NewPayment
+    )
+
+-- | Clients for "Benchmarking API".
+getHistory
+    :: Maybe (CId Wal)
+    -> Maybe CAccountId
+    -> Maybe (CId Addr)
+    -> Maybe ScrollOffset
+    -> Maybe ScrollLimit
+    -> ClientM ([CTx], Word)
+getWallets
+    :: ClientM [CWallet]
+newPayment
+    :: Maybe CPassPhrase
+    -> CAccountId
+    -> CId Addr
+    -> Coin
+    -> Maybe InputSelectionPolicy
+    -> ClientM CTx
+getHistory :<|> getWallets :<|> newPayment = client (Proxy @WalletBenchApi)
```

Comment:
Food for thoughts: I think currently this module is too coupled with the old API, which means that the day we'll decommission it we'll have to rewrite most of this module (which is fine as it's trivial) but then amend the rest of the benchmarks as the old types would have "leaked" everywhere.

I propose, as a  starter, to make the return type of these functions more opaque, i.e. let's change them to return simply a `ClientM Aeson.Value`. This way we can still inspect the blob we get from the server, if we need it (for example via `lens-aeson`) but now the code will be future-proofed. What do you think?

Reply:
Agree.